### PR TITLE
add support to hide config settings

### PR DESCRIPTION
**Pull request by JessicaGreben**
What/Why: 
An outcome from 2019-05-24's arch review is that we decided we want to simply hide RS number fields from users of uplinks. This PR extends the config system to support "hidden" config, and marks the RS fields as hidden in the uplink config.

Reference: See Jira ticket for details: https://storjlabs.atlassian.net/browse/V3-1840

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Could the PR be broken into smaller PRs?
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?

### CODE REVIEW

File: pkg/process/exec_conf_test.go [hunk 1]
```diff
@@ -44,3 +45,33 @@ func TestExec_PropagatesSettings(t *testing.T) {
...
+	// Define a config struct and some flags.
+	var config struct {
+		X int `default:"0" hidden:"true"`
+		Y int `releaseDefault:"1" devDefault:"0" hidden:"true"`
+		Z int `default:"1"`
+	}
+	Bind(cmd, &config)
+
+	// Setup test config file
+	testConfigFile, err := ioutil.TempFile("", "prefix")
```
Comment (egonelbre) [inline pkg/process/exec_conf_test.go]: Use testcontext for creating files it handles proper cleanup for you.

File: pkg/cfgstruct/bind_test.go [hunk 2]
```diff
@@ -150,3 +150,43 @@ func TestBindDevDefaults(t *testing.T) {
...
 	assertEqual(c.Fields[3].AnotherInt, int(1))
 }
+
+func TestHiddenDev(t *testing.T) {
+	f := pflag.NewFlagSet("test", pflag.PanicOnError)
+	var c struct {
+		String string `default:"dev" hidden:"true"`
+		Bool   bool   `releaseDefault:"false" devDefault:"true" hidden:"true"`
+		Int64  int64  `releaseDefault:"0" devDefault:"1"`
+		Int    int    `default:"2"`
```
Comment (ifraixedes) [inline pkg/cfgstruct/bind_test.go]: Although probably works, I would add a field with `hidden:"false"` for being also test it.

File: pkg/cfgstruct/bind_test.go [hunk 3]
```diff
@@ -150,3 +150,43 @@ func TestBindDevDefaults(t *testing.T) {
...
+	assertEqual(flagInt.Hidden, false)
+}
+
+func TestHiddenRelease(t *testing.T) {
+	f := pflag.NewFlagSet("test", pflag.PanicOnError)
+	var c struct {
+		String string `default:"dev" hidden:"true"`
+		Bool   bool   `releaseDefault:"false" devDefault:"true" hidden:"true"`
+		Int64  int64  `releaseDefault:"0" devDefault:"1"`
+		Int    int    `default:"2"`
```
Comment (ifraixedes) [inline pkg/cfgstruct/bind_test.go]: Same than above, I would add a field with `hidden:"false"`

File: pkg/process/exec_conf_test.go [hunk 4]
```diff
@@ -44,3 +45,33 @@ func TestExec_PropagatesSettings(t *testing.T) {
...
+
+func TestHidden(t *testing.T) {
+	// Set up a command that does nothing.
+	cmd := &cobra.Command{RunE: func(cmd *cobra.Command, args []string) error { return nil }}
+
+	// Define a config struct and some flags.
+	var config struct {
+		X int `default:"0" hidden:"true"`
+		Y int `releaseDefault:"1" devDefault:"0" hidden:"true"`
+		Z int `default:"1"`
```
Comment (ifraixedes) [inline pkg/process/exec_conf_test.go]: Same than above, I would add a field with `hidden:"false"`

File: pkg/cfgstruct/bind_test.go (hunk 2 above)
Comment (JessicaGreben) [reply pkg/cfgstruct/bind_test.go]: good idea. thanks.

**Pull request merged**